### ISSUE TITLE
Code Contribution: 【Hard】【Operator Development】avg_pool2d

### ISSUE THREAD

**Issue by 0x45f** [FlagOpen/FlagGems#894]
## Description 任务介绍
Develop forward function for avg_pool2d operator.
开发avg_pool2d算子的前向功能。
## Requirements 任务要求
__Interface 接口__
`avg_pool2d(Tensor self, int[2] kernel_size, int[2] stride=[], int[2] padding=0, bool ceil_mode=False, bool count_include_pad=True, int? divisor_override=None) -> Tensor`
__Function reference 功能参考__
https://docs.pytorch.org/docs/stable/generated/torch.nn.functional.avg_pool2d.html
__Implementation reference 实现参考__
https://github.com/FlagOpen/FlagGems/blob/master/src/flag_gems/ops/upsample_nearest2d.py

The operator should support all optional arguments defined in the interface. 
算子应支持接口中定义的所有参数选项。
## DDL 提交时间
Please submit a Pull Request within 2 weeks after accepting the assignment.
请于接取任务后两周内提交PR。
Please provide both accuracy test and performance test code.
请同时提供实现正确性测试与性能测试代码。

**Comment by StrongSpoon** [comment]
@Blury233

**Comment by IdleIdiot** [comment]
idiot 认领

**Comment by w1120029931-bit** [comment]
Has this operator been developed? Can it be closed?
